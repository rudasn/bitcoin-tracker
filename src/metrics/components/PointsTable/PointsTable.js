import React from 'react'
import propTypes from 'prop-types'

/**
 * Renders a table with the rows passed.
 *
 * @param {object} props
 * @param {array} props.rows An array of rows where each row
 *                           is an array containing the cell values.
 *                           Treats the first row as the header row.
 * @returns {ReactElement} PointsTable
 */
const PointsTable = ({ rows=[[]] }) =>
  <table>
    <thead>
      <tr>
      {
        rows[0].map(
          cell => <th key={ cell }>{ cell }</th>
        )
      }
      </tr>
    </thead>
    <tbody>
    {
      rows.slice(1).map(
        row => (
          <tr key={ row.join('-') }>
            {
              row.map(
                (cell, i) => <td key={ [ cell, i ].join('-') }>{ cell }</td>
              )
            }
          </tr>
        )
      )
    }
    </tbody>
  </table>

PointsTable.propTypes = {
  rows: propTypes.arrayOf(propTypes.array),
}

export default PointsTable
