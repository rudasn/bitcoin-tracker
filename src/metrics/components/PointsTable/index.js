import React from 'react'

export default ({ rows=[] }) =>
  <table>
    <thead>
      <tr>
      {
        rows[0].map(
          cell => <th key={ cell }>{ cell }</th>
        )
      }
      </tr>
    </thead>
    <tbody>
    {
      rows.slice(1).map(
        row => (
          <tr key={ row.join('-') }>
            {
              row.map(
                (cell, i) => <td key={ [ cell, i ].join('-') }>{ cell }</td>
              )
            }
          </tr>
        )
      )
    }
    </tbody>
  </table>
