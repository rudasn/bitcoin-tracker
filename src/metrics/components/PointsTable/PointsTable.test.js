import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import PointsTable from './PointsTable'

it('renders without crashing', () => {
  const element = <PointsTable />
  const div = document.createElement('div')
  ReactDOM.render(element, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders cells correctly', () => {
  const rows = [
    [ 'column1', 'column2', 'column3', 'column4' ],
    [ 0, 0, '1', '1',  ]
  ]
  const element = <PointsTable rows={ rows } />
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})
