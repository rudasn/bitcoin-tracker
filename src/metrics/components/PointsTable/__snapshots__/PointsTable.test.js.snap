// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cells correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
        column1
      </th>
      <th>
        column2
      </th>
      <th>
        column3
      </th>
      <th>
        column4
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        0
      </td>
      <td>
        0
      </td>
      <td>
        1
      </td>
      <td>
        1
      </td>
    </tr>
  </tbody>
</table>
`;
